vllm/model_executor/layers/quantization/fp8.py

```diff
@@ -528,6 +558,9 @@ def process_weights_after_loading(self, layer: Module) -> None:
         # TODO (rob): refactor block quant into separate class.
         if self.block_quant:
             assert self.quant_config.activation_scheme == "dynamic"
+            if current_platform.is_hpu():
+                layer = hpu_ops.fp8_block_moe_prepare_weights(layer)
```

Comment:
@yiliu30 , I wrapped your original set_weight funcs to vllm-hpu-extension and call it here.
BTW, I saw you put it at create_weight instead of process_weights_after_loading, I think we should put it here since we want to set final loaded weight, right? Please check if this change makes sense to you.

Reply:
Agree, we should put it at `process_weights_after_loading`, thanks!